docs/rfc/009-bit-byte-conversions.md

```diff
@@ -8,237 +8,330 @@ The Aleo Team.
+by adding 3 most significant zero bits to the bit sequence prior to grouping it into 8-bit chunks,
+leading to `M = 32` bytes, where the 3 high bits of the most significant byte are always 0.
+
+Every field element can be converted to a (little or big endian) sequence of `M` bytes,
+but, similarly to the conversions from bits above,
+attempting to convert some sequences of `M` bytes to a field element causes an error,
+precisely when the numeric value is greater than or equal to the prime number.
+
+Examples:
+```ts
+let x:field = 3;
+console.assert(x.to_bits_le() == [true, true, ...[false; 251]]);
+console.assert(x.to_bits_be() == [...[false; 251], true, true]);
+console.assert(x.to_bytes_le() == [3, ...[0; 31]];
+console.assert(x.to_bytes_be() == [...[0; 31], 3];
+let y:field = field::from_bits_le([true; 253]); // error
+let y:field = field::from_bits_be([true; 253]); // error
+let y:field = field::from_bytes_le([255; 32]); // error
+let y:field = field::from_bytes_be([255; 32]); // error
+```
 
-Given the relatively large number and regular structure of the functions above,
-it makes sense to generate them programmatically (e.g. via Rust macros),
-rather than enumerating all of them explicitly in the implementation.
-It may also makes sense, at R1CS generation time,
-to use generated or suitably parameterized code to recognize them and turn them into the corresponding gadgets.
+### Conversions with Group Elements
+
+A group element consists of two field elements that satisfy the elliptic curve equation,
+i.e. a group element is a point on the curve on the plane defined by the cartesian product of the field with itself.
+The point consists of an _x_ and a _y_ coordinate.
+
+A group element is converted to a sequence of `N` bits,
+where `N` is twice the bit length of the prime number that defines the field,
+by juxtaposing the `N/2` bits obtained from converting the _x_ and _y_ coordinates as field elements.
+More precisely:
+* The conversion of a group element (_x_, _y_) to big endian bits is `[...x, ...y]`,
+  where `x` is the conversion of _x_ to big endian bits and `y` is the conversion of _y_ to big endian bits.
+* The conversion of a group element (_x_, _y_) to little endian bits is `[...y, ...x]`,
+  where `x` is the conversion of _x_ to little endian bits and `y` is the conversion of _y_ to little endian bits.
+
+Reversing the order of the coordinates for little endian means that
+the little endian conversion of a group element is the reverse of the big endian conversion of the same group element,
+a property shared with the integer and field element conversions.
+However, given that a group element is not a single number,
+the notion of big and little endian as such does not directly apply to group elements,
+but only to their individual coordinates.
+
+Not all (little or big endian) sequences of `N` bits represent group elements.
+Not only each `N/2`-bit half must represent a field element (see discussion for field elements above),
```

Comment:
"Not only each ... must represent" --> "Not only must each ... represent

Reply:
Done.